docs/src/pipe.rst

```diff
@@ -113,3 +146,20 @@ API
     function is blocking.
 
     .. versionadded:: 1.16.0
+
+.. c:function:: int uv_pipe_get_creds(uv_pipe_t* handle, uv_pipe_creds_t* creds)
+
+    Retrieves the remote PID and effective UID/GID of a connected pipe.
+
+    On Windows, only the ``pid`` field of the credentials object is populated; ``euid``
```

Comment:
Is it worthwhile to mention that it may be possible to retrieve that information:
https://stackoverflow.com/questions/2686096/c-get-username-from-process

On Windows, it seems this would be represented by a `PSID` (aka a pointer-sized opaque object). Perhaps we should consider declaring this field as a pointer-sized word? (With either `&WinNullSid` or `(PSID)-1` used as a sentinel to indicate the value could not be retrieved.

Reply:
Actually, that might be a terrible idea, since it'd be recommending a security vulnerability. I found that MSDN has an article about how to extract the ``euid`` / ``egid`` (aka PSID) at https://docs.microsoft.com/en-us/windows/win32/SecAuthZ/verifying-client-access-with-acls-in-c--

Reply:
@vtjnash Kind of confused; I'm no windows security API expert, so I'm a little lost on the proposal here. I don't see how either of the linked articles could help.

I _did_ find a [GetSecurityInfo](https://docs.microsoft.com/en-us/windows/win32/api/aclapi/nf-aclapi-getsecurityinfo) function that would return the user/group `SID` information. Consuming applications would then just have to call [EqualSid](https://docs.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-equalsid) to check if another, separately-acquired `SID` object matches (presumably).

I'd really like an authoritative response on this, though, as I also don't want to be introducing any security holes here.

Documentation suggests that `GetSecurityInfo` could also be used with named pipes (which IIRC is what libuv uses under the hood for `uv_stream_t` objects), though I'm not sure if it would return the server user/group SIDs or the peer's.

Anyone know a MS-sec developer we could ping? ^^

Reply:
I think GetSecurityInfo on the pipe would probably return that of your end. Whereas impersonating the client, then asking about the security info of the current thread token would tell about the other end. So instead of the AccessCheck in the example, you can ask any other question about your thread in that context:

https://docs.microsoft.com/en-us/windows/win32/secauthz/access-tokens

Which I think in this case would be to call GetTokenInformation to get the TokenUser and TokenPrimaryGroup. (I think it's not GetSecurityInfo, since we have a TOKEN not a HANDLE)

But I agree it'd be helpful to have someone with authority confirm this, since I'm just taking shots in the dark.